### ISSUE TITLE
Unable to save concatenated video

### ISSUE THREAD

**Issue by hunterhawley** [pascalbaljetmedia/laravel-ffmpeg#138]
I am designing a part of my application that will allow a user to upload up to, say, 30 small video clips, and concatenate them into one larger one. I've got everything work on this except for the concatenation. I can upload each video clip to S3, convert them all to MP4's, and then delete the original files. Then when it comes time to take my converted clips and concatenate them, I run into issues. I've looked at #112 and #42, but with both of these the solution has been to name variables $video1 and $video2, which will not work for me, because I could have anywhere from 1 to 30 videos. I have attached my controller code below, is there a way that I can concatenate using the array that I already have generated, instead of naming variables for each (I could always do some dynamic variable naming, but it seems that should not be needed here).

```
$files = $request->file('video_file');
        $fileNameArrays = array();

        $name = 'video/game_date-' . $game_date . 'game_id-' . $getGameID;
        $tabCounter = 0;
        foreach($files as $theFile) {
            $name = 'video/game_date-' . $game_date . 'game_id-' . $getGameID;
            $extension = $theFile->getClientOriginalExtension();
            $theFile = file_get_contents($theFile);
            $name = $name . "_" . $tabCounter;
            Storage::disk('s3')->put($name . '.' . $extension, $theFile, 'public');

            FFMpeg::fromDisk('s3')
              ->open($name . '.' . $extension)
              ->export()
              ->toDisk('s3')
              ->inFormat(new \FFMpeg\Format\Video\X264)
              ->save($name . 'conv.mp4');
            Storage::disk('s3')->delete($name . '.' . $extension);

            array_push($fileNameArrays, $name . 'conv.mp4');

            $tabCounter += 1;
        }

        $ffmpegDisk = FFMpeg::fromDisk('s3');
        $filepath = "videos/output.mp4";
        $output = Storage::disk('s3')->getDriver()->getAdapter()->getPathPrefix() . $filepath;

        $ffmpegDisk->open($name . 'conv.mp4')->concat($fileNameArrays)->saveFromSameCodecs($output, true);
```

**Comment by hunterhawley** [comment]
Here is the error I am getting:
ffmpeg failed to execute command '/usr/local/Cellar/ffmpeg/4.1.3_1/bin/ffmpeg' '-f' 'concat' '-safe' '0' '-i' '/private/var/folders/n1/855th49s1wv63jy3qkfpb1s00000gn/T/ffmpeg-concatjpLTVp' '-c' 'copy' 'videos/output.mp4'

**Comment by hunterhawley** [comment]
Tested something new with the whole 'just 2 videos' assumption. Code below. Same error:

```
$disk = FFMpeg::fromDisk('s3');
        for($i = 1; $i <= 2; $i++) {
          ${"video$i"} = $disk->open('video/game_date-' . $game_date . 'game_id-' . $getGameID . '_' . $i . 'conv' . '.mp4');
        }
        //$output = $disk->open('/video/output_game_id_' . $getGameID . '.mp4');

        $video1->concat([
            $video1->getPathfile(),
            $video2->getPathfile(),
        ])->saveFromSameCodecs('/video/output_game_id_' . $getGameID . '.mp4', true);
```

**Comment by edlugome** [comment]
@hunterhawley did you find any solution to this issue?

**Comment by pascalbaljet** [comment]
I've rewritten the package and support for concatenation is included. Could you test the `v7` branch? Almost no breaking changes except for some filter stuff in HLS exports.

https://github.com/pascalbaljetmedia/laravel-ffmpeg/tree/v7

`composer require "pbmedia/laravel-ffmpeg:v7.x-dev"`

```php
<?php

// without transcoding
FFMpeg::fromDisk('local')
    ->open(['video.mp4', 'video2.mp4'])
    ->export()
    ->concatWithoutTranscoding()
    ->save('concat.mp4');

// with transcoding
FFMpeg::fromDisk('local')
    ->open(['video.mp4', 'video2.mp4'])
    ->export()
    ->inFormat(new X264)
    ->concatWithTranscoding($hasVideo = true, $hasAudio = true)
    ->save('concat.mp4');
```

**Comment by pascalbaljet** [comment]
Fixed in `v7.0`
